jina/peapods/pods/config/k8slib/kubernetes_deployment.py

```diff
@@ -109,87 +98,51 @@ def deploy_service(
-        template_name = 'deployment-init'
-        deployment_params = {**deployment_params, **init_container}
-    elif image_name_uses_before and image_name_uses_after:
+    template_name = 'deployment'
+
+    if image_name_uses_before and image_name_uses_after:
         template_name = 'deployment-uses-before-after'
     elif image_name_uses_before:
         template_name = 'deployment-uses-before'
     elif image_name_uses_after:
         template_name = 'deployment-uses-after'
-    else:
-        template_name = 'deployment'
-
-    if replace_deployment:
-        logger.debug(
-            f'🔋\tReplace Deployment for "{name}" with exposed port "{port_expose}"'
-        )
-        kubernetes_tools.replace(
-            deployment_name=name,
-            namespace_name=namespace,
-            template=template_name,
-            params=deployment_params,
-            custom_resource_dir=custom_resource_dir,
-        )
-        return f'{name}.{namespace}.svc'
-
-    logger.debug(f'🔋\tCreate Service for "{name}" with exposed port "{port_expose}"')
-    kubernetes_tools.create(
-        'service',
-        {
-            'name': name,
-            'target': name,
-            'namespace': namespace,
-            'port_expose': port_expose,
-            'port_in': port_in,
-            'type': 'ClusterIP',
-        },
-        logger=logger,
-        custom_resource_dir=custom_resource_dir,
-    )
-
-    logger.debug(f'📝\tCreate ConfigMap for deployment.')
-
-    kubernetes_tools.create(
-        'configmap',
-        {
-            'name': name,
-            'namespace': namespace,
-            'data': env,
-        },
-        logger=logger,
-        custom_resource_dir=None,
-    )
-
-    logger.debug(
-        f'🐳\tCreate Deployment for "{name}" with image "{image_name}", replicas {replicas} and init_container {init_container is not None}'
-    )
 
-    kubernetes_tools.create(
-        template_name,
-        deployment_params,
-        logger=logger,
-        custom_resource_dir=custom_resource_dir,
-    )
-
-    logger.debug(f'🔑\tCreate necessary permissions"')
-
-    kubernetes_tools.create(
-        'connection-pool-role',
-        {
-            'namespace': namespace,
-        },
-    )
-
-    kubernetes_tools.create(
-        'connection-pool-role-binding',
-        {
-            'namespace': namespace,
-        },
-    )
+    yamls = [
+        kubernetes_tools.get_yaml(
+            'connection-pool-role',
+            {
+                'namespace': namespace,
+            },
+        ),
+        kubernetes_tools.get_yaml(
+            'connection-pool-role-binding',
+            {
+                'namespace': namespace,
+            },
+        ),
+        kubernetes_tools.get_yaml(
+            'configmap',
+            {
+                'name': name,
+                'namespace': namespace,
+                'data': env,
+            },
+        ),
+        kubernetes_tools.get_yaml(
+            'service',
```

Comment:
Now that I am looking at it, we may not need the service in most cases. But we should double check. Maybe lets create a follow up issue for this? I think usually we only need the service actually for the gateway

Reply:
lets create a ticket for this pls

Reply:
just for completeness , the issue you created is [here](https://github.com/jina-ai/jina/issues/4144)